### ISSUE TITLE
Feature: Consider porting typescript-eslint/sort-type-constituents' groupOrder option

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [azat-io/eslint-plugin-perfectionist#151]
### What rule do you want to change?

sort-intersection-types, sort-union-types

### Describe the problem

From https://typescript-eslint.io/rules/sort-type-constituents/#grouporder: the old, feature frozen, soon-to-be-deprecated [`@typescript-eslint/sort-type-constituents` rule](https://typescript-eslint.io/rules/sort-type-constituents`) allows modifying the `groupOrder` that constituents are grouped into.

### Code example

https://typescript-eslint.io/play/#eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6ZAe2jqNNrI5Nk%2BSvliIWqDAG1w2HImjQukADRz5WedkgBzZbGIB5aABNF6MLO3bIjfIoCG8NRptQmCRsgAWkN-IAugFgAL5uwRDhoUA&code=C4TwDgpgBAogHgQwLZgDYQOICcD2BXMKAXigDs9VUoAfKARgCYBmAbiA&fileType=.ts

```ts
// groupOrder: "literal", "nullish"
type ExampleGroup = null | 123; // ❌
```

Is that something you'd want in eslint-plugin-perfectionist?

### Additional comments

This issue is coming over from https://github.com/typescript-eslint/typescript-eslint/issues/8915: wherein we discussed deprecating [`@typescript-eslint/sort-type-constituents` rule](https://typescript-eslint.io/rules/sort-type-constituents`). The only option it has that eslint-plugin-perfectionist doesn't is `groupOrder`.

But:

> But, [searching for `groupOrder` on Sourcegraph](https://sourcegraph.com/search?q=context:global+file:eslint+groupOrder+-path:.%2Bnode_modules.*+-repo:%5Egithub%5C.com/typescript-eslint/typescript-eslint%24++-path:node_modules+-lang:markdown+-file:.%2B%5C.d%5C.ts+-file:.%2B%5C.txt&patternType=keyword&sm=0) shows exactly ~two public~ very few repositories that seem to use it. Which makes me think it's not a very popular option.

Personally, I really don't think this option is worth the maintenance cost. -1 from me on this issue. The option seems very nitpicky for very little user benefit. But I figured I'd file this issue to have something tracking on the eslint-plugin-perfectionist side. 😄 

### Validations

- [X] Read the [docs](https://eslint-plugin-perfectionist.azat.io).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by tetarchus** [comment]
The thing with `groupOrder` is that it provides a quite sane default set, so many people may have come to rely on it out of the box without changing the configuration. 

As it stands, it seems that the `perfectionist` rules do not have this at all and only care about alpha/natural/line-length sorting, so the rules here report differently to how they did in the `@typescript-eslint` plugin. Due to this it likely won't just be a drop-in replacement when it is dropped from `@typescript-eslint`.

**Comment by azat-io** [comment]
@tetarchus We plan to implement this soon

**Comment by azat-io** [comment]
Thank you for your issue!

This feature was implemented in c69f27786818ecbdb881c25627ebf00eb359dbd4 and 208a9db40e17b08edbba1b4e520b5bbc2df50f8f.

All of this was released in [v3.0.0](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v3.0.0).

If you'd like, you can support the release with a retweet:
https://x.com/azat_io_en/status/1815367279191761054
